net/peer.go

```diff
@@ -907,28 +927,59 @@ func (p *Peer) RemoveP2PCollections(collections []string) error {
 		}
 	}
 
-	return txn.Commit(p.ctx)
+	return nil
 }
 
-// GetAllP2PCollections gets all the collectionIDs from the pubsup topics
-func (p *Peer) GetAllP2PCollections() ([]client.P2PCollection, error) {
+// GetAllP2PCollectionsFromStore gets all the collectionIDs that have been added to the
+// pubsub topics from the system store.
+func (p *Peer) GetAllP2PCollectionsFromStore() ([]client.P2PCollection, error) {
 	txn, err := p.db.NewTxn(p.ctx, false)
 	if err != nil {
 		return nil, err
 	}
+	defer txn.Discard(p.ctx)
 	store := p.db.WithTxn(txn)
 
 	collections, err := p.db.GetAllP2PCollections(p.ctx)
 	if err != nil {
-		txn.Discard(p.ctx)
 		return nil, err
 	}
 
 	p2pCols := []client.P2PCollection{}
 	for _, colID := range collections {
 		col, err := store.GetCollectionBySchemaID(p.ctx, colID)
 		if err != nil {
-			txn.Discard(p.ctx)
+			return nil, err
+		}
+		p2pCols = append(p2pCols, client.P2PCollection{
+			ID:   colID,
+			Name: col.Name(),
+		})
+	}
+
+	return p2pCols, txn.Commit(p.ctx)
+}
+
+// GetAllP2PCollectionsFromServer gets all the collectionIDs that have been added to the
+// pubsub topics from the system store.
```

Comment:
todo: update comment

2 methods `GetAllP2PCollectionsFromStore` and `GetAllP2PCollectionsFromServer` have the same comment